src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -152,6 +166,7 @@ public override void Initialize(AnalysisContext context)
                 }
 
                 ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols = new();
+                ConcurrentDictionary<ISymbol, ValueTuple<string?, string?>> previewSymbolToMessageAndUrl = new();
```

Comment:
Why did not use Could not use the above dictionary? It seems to me now we have cache with similar items. I guess that is what you meant by `I could fold the new ConcurrentDictionary into the existing SymbolIsAnnotatedAsPreview call`,  for me that looks better, up to you

Reply:
Working on that now.